app/lib/request.rb

```diff
@@ -127,21 +128,24 @@ def body_with_limit(limit = 1.megabyte)
   end
 
   class Socket < TCPSocket
-    class << self
-      def open(host, *args)
-        outer_e = nil
-        Addrinfo.foreach(host, nil, nil, :SOCK_STREAM) do |address|
-          begin
-            raise Mastodon::HostValidationError if PrivateAddressCheck.private_address? IPAddr.new(address.ip_address)
-            return super address.ip_address, *args
-          rescue => e
-            outer_e = e
+    if Rails.configuration.x.http_client_proxy.empty?
```

Comment:
Seems like this conditional can be avoided by conditionally setting the `socket_class` option in initialize instead